File: CS/E1554.Module/Module.cs
using System;
using System.Collections.Generic;

using DevExpress.ExpressApp;
using System.Reflection;
using DevExpress.ExpressApp.Model;
using System.ComponentModel;


namespace E1554.Module {
    public sealed partial class E1554Module : ModuleBase {
        public E1554Module() {
            InitializeComponent();
        }
        public override void ExtendModelInterfaces(ModelInterfaceExtenders extenders) {
            base.ExtendModelInterfaces(extenders);
            extenders.Add<IModelListView, IModelListViewAdditionalCriteria>();
        }
    }
    public interface IModelListViewAdditionalCriteria : IModelNode {
        [DefaultValue("")]
        string AdditionalCriteria { get; set; }
    }
}

File: VB/E1554.Module/Module.vb
Imports System
Imports System.Collections.Generic

Imports DevExpress.ExpressApp
Imports System.Reflection
Imports DevExpress.ExpressApp.Model
Imports System.ComponentModel


Namespace E1554.Module
	Public NotInheritable Partial Class E1554Dodule
		Inherits ModuleBase

		Public Sub New()
			InitializeComponent()
		End Sub
		Public Overrides Sub ExtendModelInterfaces(ByVal extenders As ModelInterfaceExtenders)
			MyBase.ExtendModelInterfaces(extenders)
			extenders.Add(Of IModelListView, IModelListViewAdditionalCriteria)()
		End Sub
	End Class
	Public Interface IModelListViewAdditionalCriteria
		Inherits IModelNode

		<DefaultValue("")>
		Property AdditionalCriteria() As String
	End Interface
End Namespace
